newsroom/notifications/utils.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+
+from bson import ObjectId
+import superdesk
+
+from typing import Any
+from superdesk.utc import utcnow
+from superdesk.flask import session
+from superdesk.core import get_app_config
+from superdesk.notification import push_notification
+
+from .services import NotificationsService
+
+
+def user_notifications_lookup(user_id: str | ObjectId) -> dict[str, Any]:
+    if isinstance(user_id, str):
+        user_id = ObjectId(user_id)
+
+    ttl = get_app_config("NOTIFICATIONS_TTL", 1)
+    return {
+        "user": user_id,
+        "_created": {"$gte": utcnow() - datetime.timedelta(days=ttl)},
+    }
+
+
+async def get_user_notifications(user_id: str) -> list[dict[str, Any]]:
+    """
+    Returns the notification entries for the given user
+    """
+    lookup = user_notifications_lookup(user_id)
+    cursor = await NotificationsService().search(lookup)
+    return await cursor.to_list_raw()
+
+
+async def get_initial_notifications() -> dict[str, Any] | None:
+    """
+    Returns the stories that user has notifications for
+    :return: List of stories. None if there is not user session.
+    """
+    if not session.get("user"):
+        return None
+
+    cursor = await NotificationsService().search(lookup=user_notifications_lookup(session["user"]))
+
+    return {
+        "user": str(session["user"]) if session["user"] else None,
```

Comment:
We probably don't need a second if condition for `session["user"]` when we're already providing one at the top of this function (I fear I might have been the one that added this... :face_with_head_bandage: )